shotgun_api3/shotgun.py

```diff
@@ -1752,8 +1794,24 @@ def _response_errors(self, sg_response):
         """
 
         if isinstance(sg_response, dict) and sg_response.get("exception"):
-            raise Fault(sg_response.get("message",
-                "Unknown Error"))
+            
+            # handle the following cases explicitly
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Can't authenticate session token 'xxxxxxxx'"}
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Session token 'xxxxxxxx has expired'"}
+            
+            msg = sg_response.get("message")
+            
+            if msg and ("Can't authenticate session token" in msg or "has expired" in msg):
+                raise SessionAuthenticationFault(msg)
```

Comment:
The errors have an error code as well - i think the problem here is that *different* error messages coming from the server are using the *same* error code - where probably each error should have been using a different error code (at least if we wanted to use the codes for raising better exceptions on the client side). I am guessing it is too late to change on the server side too - that may risk breaking something else... Or do you think there is some good way to handle both scenarios?